### PR TITLE
solutoin added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public List<Order> getAll() {
+        return Storage.orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        IntStream.range(0, Storage.orders.size())
+                .filter(i -> Storage.orders.get(i).getOrderId().equals(order.getOrderId()))
+                .forEach(i -> Storage.orders.set(i, order));
+
```
Comment (aleksandr-hrankin) [inline src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java]: empty line

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        cart.getProducts().clear();
+        shoppingCart.update(cart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCart.getAll().stream()
+                .filter(cart -> cart.getUserId().equals(userId))
+                .findFirst()
+                .orElseThrow();
```
Comment (aleksandr-hrankin) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: Эту логику необходимо вынести в Dao слой

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    List<Order> getAll();
+
+    Order update(Order order);
+
+    boolean deleteById(Long id);
+
+    boolean delete(Order order);
+
+    public Order completeOrder(ShoppingCart shoppingCart);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java]: this is not CRUD operation, let's move it to service layer

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart cart, Product product) {
+        cart.getProducts().add(product);
+        return shoppingCart.update(cart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart cart, Product product) {
+        return cart.getProducts().remove(product);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: you forgot to update shoppingCart after you changed its field products

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    List<Order> getAll();
+
+    Order update(Order order);
+
+    boolean deleteById(Long id);
+
+    boolean delete(Order order);
+
+    public List<Order> getUserOrders(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java]:
```suggestion
   List<Order> getUserOrders(Long userId);
```
public is access modifier by default in interfaces

File: src/main/java/com/internet/shop/dao/ShoppingCartDao.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    List<ShoppingCart> getAll();
+
+    ShoppingCart update(ShoppingCart cart);
+
+    boolean deleteById(Long id);
+
+    boolean delete(ShoppingCart cart);
+
+    public Optional<ShoppingCart> getByUserId(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/ShoppingCartDao.java]: same here

File: src/main/java/com/internet/shop/service/OrderService.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package com.internet.shop.service;
+
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+
+public interface OrderService {
+    Order create(Order order);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/OrderService.java]: we don't need create() method on service layer
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/homework/lesson_14_hw2.html

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**